Skip Advancement checks for ManaItem updates

jtuc opened the pull request:
Advancement checking is expensive, and mana items will often update every tick.

I don't think this will interfere with any actual advancements? The bauble advancement uses and ImpossibleTrigger and is manually granted in the BaubleItem onEquipped method. Worst case scenario, the advancement will just be delayed until a non-mana item triggers the check.

Should address #4793

jtuc committed:
Skip Advancement checks for ManaItem updates

Advancement checking is expensive, and mana items will often update
every tick.

jtuc commented on Xplat/src/main/java/vazkii/botania/mixin/ServerPlayerMixin.java:
```diff
@@ -39,4 +41,14 @@ private void onTick(CallbackInfo ci) {
+		@Inject(at = @At("HEAD"), method = "slotChanged", cancellable = true)
+		private void onSlotChanged(AbstractContainerMenu containerMenu, int slotIndex, ItemStack itemStack, CallbackInfo ci) {
+			// ManaItems update frequently - skip advancement processing for them
+			if (XplatAbstractions.INSTANCE.findManaItem(itemStack) != null) {
```
Possibly worth hardcoding item classes instead for more performance? Not sure how annoying that list would be to maintain...

TheRealWormbo replied:
Ideally this would only skip the check if the item type did not change, as the same system is also responsible for unlocking recipes.

jtuc replied:
I don't think there's a good way to check the prior inventory - at the point this listener is invoked, the inventory has already been updated, so we can't tell if it was just an NBT change or a new item.

Since only the manaTablet triggers recipe unlocks, I think special-casing the first manaTablet per player resolves this?
If modpacks/addons add different botania advancements/recipes, they'll still be triggered the next time a player moves a non-manaItem in their inventory.

jtuc committed:
Run Advancement triggers for first manaTablet

The manaTablet is used in recipe unlock advancements, so we should let
it run triggers at least once per player.

jtuc commented:
As discussed on #4793, it probably makes more sense for users to install a generic achievements optimizing mod instead of each mod having their own (potentially conflicting) mixins.
